h2/src/main/org/h2/value/ValueArray.java

```diff
@@ -24,9 +23,9 @@
      */
     private static final Object EMPTY = get(new Value[0]);
 
-    private final Class<?> componentType;
+    private volatile TypeInfo componentType;
```

Comment:
there is no need for this to be volatile since the worst that can happen is that we do some unneccasry recalculation

Reply:
Some JRE's may write an incompletely initialized object to non-volatile field due to reordering of operations. However, here the right side of assignment is not a constructor and it is created by another method that can't be inlined due to its size, so I think that this object should be already initialized and the modifier may be removed.

Reply:
#2189